Source/OxyPlot/Series/HeatMapSeries.cs

```diff
@@ -172,6 +187,58 @@ public void Invalidate()
         }
 
         /// <summary>
+        /// Gets the clipping rectangle, transposed if the X acis is vertically orientated
+        /// </summary>
+        protected new OxyRect GetClippingRect()
+        {
+            double minX = Math.Min(this.XAxis.ScreenMin.X, this.XAxis.ScreenMax.X);
+            double minY = Math.Min(this.YAxis.ScreenMin.Y, this.YAxis.ScreenMax.Y);
+            double maxX = Math.Max(this.XAxis.ScreenMin.X, this.XAxis.ScreenMax.X);
+            double maxY = Math.Max(this.YAxis.ScreenMin.Y, this.YAxis.ScreenMax.Y);
+
+            if (XAxis.IsVertical())
+                return new OxyRect(minY, minX, maxY - minY, maxX - minX);
+            else
+                return new OxyRect(minX, minY, maxX - minX, maxY - minY);
+        }
+
+        /// <summary>
+        /// Transposes the ScreenPoint if the X axis is vertically orientated
+        /// </summary>
+        private ScreenPoint Orientate(ScreenPoint point)
+        {
+            if (XAxis.IsVertical())
+            {
+                point = new ScreenPoint(point.Y, point.X);
+            }
+            return point;
+        }
+
+        /// <summary>
+        /// Transforms DataSpace coords to orientated ScreenSpace coors
+        /// </summary>
+        public new ScreenPoint Transform(double x, double y)
+        {
+            return Orientate(base.Transform(x, y));
+        }
+
+        /// <summary>
+        /// Transforms DataSpace coords to orientated ScreenSpace coors
+        /// </summary>
+        public new ScreenPoint Transform(DataPoint p)
+        {
+            return Orientate(base.Transform(p));
+        }
+
+        /// <summary>
+        /// Transforms orientated ScreenSpace coords to DataSpace coors
+        /// </summary>
+        public new DataPoint InverseTransform(ScreenPoint point)
+        {
+            return base.InverseTransform(Orientate(point));
+        }
```

Comment:
I did some testing today, and just copying these shadow methods into LineSeries and Column appears to make them behave as one would expect them (though I appreciate that Column charts tend to have columns, not bars) - ScatterSeries is also easy 'fixed' with these methods, but the calls to XAxis.Transform() have to replaced with calls to the shadow methods (I can't see any reason for not using the base Transform(double, double) as it is), and a similar story for AreaSeries. I shall continue to look into this, because it would be nice if this could be pushed down to XYAxisSeries or an intermediate class.